processor/sourcemap/payload.go

```diff
@@ -20,14 +21,14 @@ type payload struct {
 	BundleFilepath string `mapstructure:"bundle_filepath"`
 }
 
-func (pa *payload) transform(smapAccessor utility.SmapAccessor) []beat.Event {
+func (pa *payload) transform(config *pr.Config) []beat.Event {
 	var events = []beat.Event{pr.CreateDoc(mappings(pa))}
 	sourcemapCounter.Add(1)
 
-	if smapAccessor == nil {
+	if config == nil || config.SmapMapper == nil {
 		logp.Err("Sourcemap Accessor is nil, cache cannot be invalidated.")
 	} else {
-		smapAccessor.RemoveFromCache(utility.SmapID{
+		config.SmapMapper.NewSourcemapAdded(smap.Id{
```

Comment:
This is tricky, if the sourcemap can't be saved in ES for whatever reason, it will only be saved in the cache.
Once it expires, sourcemaps will suddenly stop working.

Quite a corner case, but 🤷‍♂️

Reply:
aahh, scratch that. that is removing it, not adding it 🤦‍♂️ 
all good then